SpringProject/src/main/java/com/gradproject2019/users/repository/UserRepository.java

```diff
@@ -54,4 +54,21 @@ void updateUser(@Param("id") Long id,
             "ORDER BY last_name ASC",
             nativeQuery = true)
     List<User> searchByName(@Param("query") String query);
+
+    @Query(value = "SELECT EXISTS(" +
+            "SELECT 1 " +
+            "FROM users " +
+            "WHERE id = :requestedUserId " +
+            "AND " +
+            "(" +
+            "(manager_id IS NOT NULL " +
+            "AND manager_id = :requestingUserId) " +
+            "OR " +
+            "(:requestingUserManagerId IS NOT NULL " +
+            "AND id = :requestingUserManagerId)" +
+            "))",
+            nativeQuery = true)
+    int hasManagerEmployeeRelationship(@Param("requestedUserId") Long requestedUserId,
+                                       @Param("requestingUserId") Long requestingUserId,
+                                       @Param("requestingUserManagerId") Long requestingUserManagerId);
```

Comment:
The manager ID of the requesting user can be inferred from the table, so it's not needed as a parameter?

Reply:
Is it not easier to just put it in as a parameter rather than having to retrieve it from the requesting user's row?

Reply:
I suppose it's easier to write, it's up to you.